### PR TITLE
Fixed Validation messages missing from datepicker form elements

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

Fixed Validation messages missing from datepicker form elements
### Fixed Issues (if relevant)

1. magento/magento2#3795: Validation messages missing from datepicker form elements

### Manual testing scenarios (*)

1.  Edit a CMS Page
2. Choose Custom Design From and Custom Design To date,  end date occurs **before** the start date.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by rogyar** [comment]
Hi @ravi-chandra3197. Please, consider fixing failing tests

**Comment by sidolov** [comment]
Hi @ravi-chandra3197 , tests still failing. Please, take a look

**Comment by sidolov** [comment]
@ravi-chandra3197 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

File: app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js [hunk 1]
```diff
@@ -803,6 +803,14 @@ define([
             },
             $.mage.__('Please enter a valid date.')
         ],
+        'validate-date-range': [
+            function (value, params) {
+                var fromDate = $('input[name*="' + params + '"]').val();
+
+                return (moment.utc(value).unix() >= moment.utc(fromDate).unix()) || isNaN(moment.utc(value).unix());
```
Comment (dmytro-ch) [inline app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js]: Please remove the redundant parentheses in order to fix the static test failure.

File: app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js [hunk 2]
```diff
@@ -803,6 +803,14 @@ define([
             },
             $.mage.__('Please enter a valid date.')
         ],
+        'validate-date-range': [
+            function (value, params) {
```
Comment (dmytro-ch) [inline app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js:807]: Shouldn't we make the validation logic for `validate-date-range` rule to be the same as in `lib/web/mage/validation.js` for better consistency?

**Review by dmytro-ch — CHANGES_REQUESTED** [review]

File: app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js (hunk 1 above)
Comment (ravi-chandra3197) [reply app/code/Magento/Ui/view/base/web/js/lib/validation/rules.js]: Hello @dmytro-ch 
I have update PR code as per your suggestion for remove the redundant parentheses to fix the static test. can you review it.

**Comment by dmytro-ch** [comment]
@ravi-chandra3197,
could you please squash all commits into a single commit in order to cleanup history?

**Comment by ravi-chandra3197** [comment]
> @ravi-chandra3197,
> could you please squash all commits into a single commit in order to cleanup history?

Hello @dmytro-ch
I have squashed all commits into a single commit. can you review it.

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-5118](https://jira.corp.magento.com/browse/ENGCOM-5118) has been created to process this Pull Request

**Comment by soleksii** [comment]
:heavy_check_mark: QA Passed

**Before:**

before

**After:**

after

**Pull request merged**